src/systemd.c

```diff
@@ -0,0 +1,272 @@
+/* SPDX-License-Identifier: LGPL-2.1-only */
+/**
+ * Copyright (c) 2022 Oracle and/or its affiliates.
+ * Author: Tom Hromatka <tom.hromatka@oracle.com>
+ * Author: Silvia Chapa <silvia.chapa@oracle.com>
+ */
+
+#include <libcgroup-internal.h>
+#include <systemd/sd-bus.h>
+#include <libcgroup.h>
+#include <unistd.h>
+#include <assert.h>
+#include <errno.h>
+
+#define ONE_SEC 1000000
+
+static const char * const modes[] = {
+	"fail",			/* CGROUP_SYSTEMD_MODE_FAIL */
+	"replace",		/* CGROUP_SYSTEMD_MODE_REPLACE */
+	"isolate",		/* CGROUP_SYSTEMD_MODE_ISOLATE */
+	"ignore-dependencies",	/* CGROUP_SYSTEMD_MODE_IGNORE_DEPS */
```

Comment:
Should we restrict our API support to `CGROUP_SYSTEMD_MODE_FAIL` and `CGROUP_SYSTEMD_MODE_REPLACE` only? `systemctl` man pages says that the `CGROUP_SYSTEMD_MODE_ISOLATE` instructs systemd to stop all other services and run this unit and its dependencies. Similarly, `CGROUP_SYSTEMD_MODE_IGNORE_DEPS` and `CGROUP_SYSTEMD_MODE_IGNORE_REQS` don't apply to the simple `slice/scope` we are creating.

Reply:
That seems reasonable.  We could always add support for them later if/when we need them.  I'll add a check for these two modes.  Thanks.

Reply:
I added this check:
```
        if (opts->mode == CGROUP_SYSTEMD_MODE_ISOLATE ||
            opts->mode == CGROUP_SYSTEMD_MODE_IGNORE_DEPS ||
            opts->mode == CGROUP_SYSTEMD_MODE_IGNORE_REQS) {
                cgroup_err("unsupported systemd mode: %d\n", opts->mode);
                return ECGINVAL;
        }
```

Reply:
looks good to me.